tests/samples/bluetooth/samples_test_app/src/main.c

```diff
@@ -0,0 +1,428 @@
+
+static struct bt_conn *default_conn;
+
+char *dev_names[] = {
+	"Nordic_UART_Service", "Nordic_Throughput", "NCS_HIDS_keyboard",
+	"NCS_HIDS_mouse",      "Nordic_LBS",	    "Test beacon",
+};
+
+char *hid_uuid = (char *)BT_UUID_HIDS;
+
+static void uart_cb(const struct device *uart, void *user_data)
+{
+	ARG_UNUSED(user_data);
+	static struct uart_data_t *rx;
+
+	uart_irq_update(uart);
+
+	if (uart_irq_rx_ready(uart)) {
+		int data_length;
+
+		if (!rx) {
+			rx = k_malloc(sizeof(*rx));
+			if (rx) {
+				rx->len = 0;
+			} else {
+				printk("Not able to allocate UART receive buffer\n");
+				return;
+			}
+		}
+
+		data_length = uart_fifo_read(uart, &rx->data[rx->len],
+					     UART_BUF_SIZE - rx->len);
+		rx->len += data_length;
+
+		if (rx->len > 0) {
+			/* Send buffer to Bluetooth unit if either buffer size
+			 * is reached or the char \n or \r is received, which
+			 * ever comes first
+			 */
+			if ((rx->len == UART_BUF_SIZE) ||
+			    (rx->data[rx->len - 1] == '\n') ||
+			    (rx->data[rx->len - 1] == '\r')) {
+				k_fifo_put(&fifo_uart_rx_data, rx);
+				rx = NULL;
+			}
+		}
+	}
+
+	if (uart_irq_tx_ready(uart)) {
+		struct uart_data_t *buf =
+			k_fifo_get(&fifo_uart_tx_data, K_NO_WAIT);
+		uint16_t written = 0;
+
+		/* Nothing in the FIFO, nothing to send */
+		if (!buf) {
+			uart_irq_tx_disable(uart);
+			return;
+		}
+
+		while (buf->len > written) {
+			written += uart_fifo_fill(uart, &buf->data[written],
+						  buf->len - written);
+		}
+
+		while (!uart_irq_tx_complete(uart)) {
+			/* Wait for the last byte to get
+			 * shifted out of the module
+			 */
+		}
+
+		if (k_fifo_is_empty(&fifo_uart_tx_data)) {
+			uart_irq_tx_disable(uart);
+		}
+
+		k_free(buf);
+	}
+}
+
+static void discovery_complete(struct bt_gatt_dm *dm, void *context)
+{
+	printk("Service discovery completed\n");
+
+	bt_gatt_dm_data_print(dm);
+
+	bt_gatt_dm_data_release(dm);
+}
+
+static void discovery_service_not_found(struct bt_conn *conn, void *context)
+{
+	printk("Service not found\n");
+	printk("FAILURE - Service not found\n");
+}
+
+static void discovery_error(struct bt_conn *conn, int err, void *context)
+{
+	printk("Error while discovering GATT database: (%d)\n", err);
+}
+
+struct bt_gatt_dm_cb discovery_cb = {
+	.completed = discovery_complete,
+	.service_not_found = discovery_service_not_found,
+	.error_found = discovery_error,
+};
+
+static void gatt_discover(struct bt_conn *conn)
+{
+	if (conn == default_conn) {
+		int err = bt_gatt_dm_start(conn, NULL, &discovery_cb, NULL);
+
+		if (err) {
+			printk("Failed to start discovery process (err:%d)\n",
+			       err);
+			printk("FAILURE - Discovery start failed\n");
+		}
+	}
+}
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		printk("Failed to connect to %s (%d)\n", addr, conn_err);
+		printk("FAILURE - Failed to connect\n");
```

Comment:
2 outputs for the same thing are not needed

Reply:
removed